### PR TITLE
#224, #226: Replacing Maven 2 APIs with Maven 3

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Replacing Maven 2 APIs like ArtifactMetadataSource, ArtifactResolver, WagonManager, etc., with current, non-deprecated alternatives. Also, moving to JSR330 to be compliant with latest Maven standards and recommendations.

`Mojo`, `Artifact`: Class name changed to be in line with maven plugin naming conventions.

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
@longtimeago please review

File: src/main/java/com/googlecode/download/maven/plugin/internal/WGetMojo.java [hunk 1]
```diff
@@ -462,20 +449,26 @@ private boolean isFileUnArchiver(final UnArchiver unarchiver) {
...
-                    .withProxyHost(proxyInfo.getHost())
-                    .withProxyPort(proxyInfo.getPort())
-                    .withProxyUserName(proxyInfo.getUserName())
-                    .withProxyPassword(proxyInfo.getPassword())
-                    .withNtlmDomain(proxyInfo.getNtlmDomain())
-                    .withNtlmHost(proxyInfo.getNtlmHost());
-        }
 
-        if (!skipCache) {
+        final RemoteRepository repository = new RemoteRepository.Builder(isBlank(this.serverId)
```
Comment (longtimeago) [inline src/main/java/com/googlecode/download/maven/plugin/internal/WGetMojo.java:455]: @ajarmoniuk this block with nested ternary operations deserves own method

**Review by longtimeago — COMMENTED** [review]
Uff, quite a lot of changes ... did my best but it's hard to catch something here.
Please, see several suggestions below.
Also, does migrating to the API means we don't support Maven 2 anymore?
If yes, please, make it clear in readme

Thank you!

File: src/it/basicArtifactTransitive/invoke.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+invoker.goals = ${project.groupId}:${project.artifactId}:${project.version}:artifact
+invoker.mavenOpts = -DgroupId=javax.annotation -DartifactId=javax.annotation-api -Dversion=1.3.2 -DdependencyDepth=1
```
[the remark this answers is not in the document]
Comment (ajarmoniuk) [reply src/it/basicArtifactTransitive/invoke.properties]: Ok. Turns out, artifact retrieval wasn't done correctly. Fixed that.

- Correcting retrieval of dependency artifacts
- Using a mock repository
- Correcting the integration test for the artifact mojo

**Review by longtimeago — APPROVED** [review]
@ajarmoniuk awesome, thanks a lot!

**Pull request merged**